tools/sof-tplgreader.py

```diff
@@ -231,16 +277,26 @@ def func_getPipeline(tplgObj, tplgName, sdcard_id, sort):
     ret_args = vars(parser.parse_args())
 
     tplgreader = clsTPLGReader()
-    filter_lst = []
+    filter_dict = {"filter":[], "op":[]}
     dump_lst = []
     ignore_lst = []
     pipeline_lst = []
     tplg_root = ""
 
     if ret_args['filter'] is not None and len(ret_args['filter']) > 0:
-        for emStr in ret_args['filter']:
-            key, flag, value=emStr.partition(':')
-            filter_lst.append({key.strip():value.strip().replace('[','').replace(']','').split(',')})
+        # parse filter string into two structures, one is dict list for each filter item,
+        # and another is logic operation list.
+        filter_lst = []
+        op_lst = []
+        for filter_elem in ret_args['filter'].split('|'):
+            for item in filter_elem.split('&'):
+                key, _, value = item.partition(':')
+                filter_lst.append({key.strip():value.strip().replace('[','').replace(']','').split(',')})
+        for char in ret_args['filter']:
+            if char == '|' or char == "&":
+                op_lst.append(char)
+        filter_dict['filter'] = filter_lst
+        filter_dict['op'] = op_lst
```

Comment:
1. `[]` value can include this?
2.  you get order `op_lst`, but `filter_lst ` is order by `'|', '&'` , I think this is will cause problem  
maybe you can try `for char in len(ret_args['filter'])`   
here will include those behavior:
```
if '|' in ret_args['filter'] or '&' in ret_args['filter']:
   ...
else:
   loop ret_args['filter']
fi
```

Reply:
for item 1, delete the replacement of '[' and ']'.